src/rust/engine/process_execution/src/lib.rs

```diff
@@ -880,6 +881,11 @@ pub trait CommandRunner: Send + Sync + Debug {
     workunit: &mut RunningWorkunit,
     req: Process,
   ) -> Result<FallibleProcessResultWithPlatform, ProcessError>;
+
+  /// Shutdown this CommandRunner cleanly. This
+  async fn shutdown(&self) -> Result<(), String> {
+    Ok(())
```

Comment:
Given how many implementations have needed to override this, it would probably be reasonable to just require it?

Reply:
The only impls that need to override are those with an inner command runner (e.g., `SwitchedCommandRunner`) or those that need explicit shutdown behavior (e.g., `docker::CommandRunner`). The "leaf" command runners like nailgun and local don't need to override it.

Reply:
Switched to having every trait impl implement it. That way, any future impls will be required to think about how they should implement it.